src/slotscheck/discovery.py

```diff
@@ -308,7 +308,7 @@ def _module_parents(
         if pp in sys_path:
             return
         yield pp
-    raise ValueError(f"File {p} is outside of PYTHONPATH ({sys.path})")
+    raise ModuleNotFoundError(f"No module named '{p.stem}'", name=p.stem)
```

Comment:
I understand that you want to reuse the logic in cli.py, but this error can be very misleading. `ModuleNotFoundError` means we could not resolve a "dotted.module.name". Here we could not find a module name for a file path. We were going in the opposite direction (file -> module instead of module -> file), which does not normally happen in Python. I suggest that we add a new exception type for this, keep the full file path, and handle it in cli.py similar to ModuleNotFoundError.

Reply:
Agree, let's do a custom exception 👍